### ISSUE TITLE
[Accessibility]: Blob view is announced as text area, screen reader cursor doesn't move within the code

### ISSUE THREAD

**Issue by felixfbecker** [sourcegraph/sourcegraph#43375]
### Audit type

Screen reader navigation

### Problem description

When trying to read the code with a screen reader, the blob view is announced as a textarea (i.e. an input box).
As code line are read, the screen reader will always repeat "You are currently on a text area" (which would be true if it was an editor, but it's not).
Additionally the screen reader cursor is always on the entire blob view (i.e., the "text area") rather than following the individual lines/words of code and the focused links as you would expect for a non-text-area (a text document), impacting the user's ability to orient themselves. This is especially noticeable when we make the tokens tab-focusable links, the cursor will stay on the outer blob view.

### Expected behavior

- The blob view should not be announced as a text area, ~~but probable be an `<article>` or other role~~ _an ancestor representing the blob is already a `region`_
- The screen reader cursor should follow the currently read line/tokens.

### Additional details

This happens because CodeMirror explicitly wraps the whole blob in a `<div role="textbox" contenteditable="false">`:

![image](https://user-images.githubusercontent.com/10532611/197611044-63b56666-28f3-4ab3-ad07-04b15f15dd39.png)

![image](https://user-images.githubusercontent.com/10532611/197611113-3efde3c1-093e-46bb-9a31-22106ad03275.png)

Removing `role="textbox"` in the DOM explorer makes it behave as expected.

cc @fkling @tomro

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by fkling** [comment]
It's possible to set `role` to an empty value... would that fix it too (I have no easy way to test atm).
Is this something we should raise upstream? Seems like if the editor is readonly maybe it should not have a `role="textbox"` by default. Or maybe ours is a special case.

**Comment by felixfbecker** [comment]
> It's possible to set role to an empty value (or any other value of that matter)... would that fix it too? (I have no easy way to test atm).

Yeah I think an empty string should work, that should be the same as no `role` at all, or the explicit role `"generic"` (default for a `<div>`). Meaning that wrapper element would just be `generic` and not show up in the accessibility tree.

> Is this something we should raise upstream? Seems like if the editor is readonly maybe it should not have a role="textbox" by default. Or maybe ours is a special case.

There is a semantic difference between a read-only textbox and just text. E.g. consider the difference between
```html
<input readonly value="Hello">
```
and
```html
<span>Hello</span>
```
So I don't think they should automatically do this. A read-only textbox would make sense for an editor where the current file may be uneditable due to not having write permissions on the file, or due to being some kind of virtual content (like an interface from an assembly/DLL). Basically I think if there are other cases within an application where the CodeMirror instance _could be_ editable, you'd want to only mark it as `readonly`, but if it's always just a viewer, you want it to just be content and not announce itself as explicitly "readonly".
